### PR TITLE
[FEATURE]:Searchbar added in blog page

### PR DESCRIPTION
**Pull request by BhartiNagpure**
## Description
#489 Searchbar added
I am added search bar in Blog page to search particular blog which user want.

## Pull Request Checklist

Please make sure that your PR meets the following requirements:

- [x] The app is working correctly after applying the changes.
- [x] The app is responsive and displays properly on different screen sizes and devices.
- [x] The changes are thoroughly documented in the appropriate files or documentation sections.
- [x] The coding conventions and style guidelines are followed consistently.

## Checklist

Please check all applicable boxes and provide any additional information if needed.

- [x] I have tested the changes locally and ensured that the app is functioning as expected.
- [x] I have verified that the app is responsive and looks good on various screen sizes and devices.
- [x] I have reviewed the documentation and ensured that it accurately reflects the changes made.
- [x] I have followed the coding conventions and style guidelines defined in the project.
- [x] I have run the automated tests and all of them pass.
- [x] I have considered the performance implications of the changes made.
- [x] I have considered the security implications of the changes made.
- [x] I have checked for any potential regressions or side effects caused by the changes.
- [x] I have performed a self-review of my code.
- [x] Starred the repository.
- [x] Added My Name to the Contributor's List.
- [x] Followed the repository's [Contributing Guidelines](/CONTRIBUTING.md).
- [x] I ran the app and tested it locally to verify that it works as expected.

# Screenshots - If Any (Optional)
Screenshot 2023-07-13 211310

## Related Issues
#489 Feature searchbar added in blog

## Reviewers

Please tag the relevant team members or reviewers who should review this pull request.

### CODE REVIEW

**Comment by BhartiNagpure** [comment]
@Harshil-Jani  please review my PR .

**Comment by Harshil-Jani** [comment]
@BhartiNagpure I tried searching but it returns no blog for me. Can you please re-check ?

File: components/Blogs/BlogCard.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faThumbsUp, faComment } from "@fortawesome/free-solid-svg-icons";
 import Link from "next/link";
 
+
+
```
Comment (Harshil-Jani) [inline components/Blogs/BlogCard.js:9]: Please remove extra whitespaces.

File: components/Blogs/BlogCard.js [hunk 2]
```diff
@@ -43,7 +45,9 @@ const BlogCard = (props) => {
             </div>
           </div>
         </div>
+
```
Comment (Harshil-Jani) [inline components/Blogs/BlogCard.js:48]: Please remove extra whitespaces.

File: components/Blogs/BlogCard.js [hunk 3]
```diff
@@ -43,7 +45,9 @@ const BlogCard = (props) => {
             </div>
           </div>
         </div>
+    
       </Link>
+
```
Comment (Harshil-Jani) [inline components/Blogs/BlogCard.js:50]: Please remove extra whitespaces.

File: pages/blog/index.js [hunk 4]
```diff
@@ -6,19 +6,45 @@ import styles from "@styles/scss/blog.module.scss";
 import blogData from "/data/blogdata";
 import GoToTop from "components/GoToTop";
 import WriteBlog from "./writeblog";
+import { useState } from "react";
+
```
Comment (Harshil-Jani) [inline pages/blog/index.js:10]: Please remove extra whitespaces.

File: pages/blog/index.js [hunk 5]
```diff
@@ -6,19 +6,45 @@ import styles from "@styles/scss/blog.module.scss";
...
 export default function Blogs() {
+	const [query, setQuery] = useState("");
+	const handleSearch=(e)=>{
+		setQuery (e.target.value)
+	}
+
+   const searchQueryHandler = () =>{
+
+   }
+   console.log(blogData.filter(blog => blog.title.includes("SQ")))
```
Comment (Harshil-Jani) [inline pages/blog/index.js:21]: Remove console.log statement.

File: pages/blog/index.js [hunk 6]
```diff
@@ -30,7 +56,7 @@ export default function Blogs() {
 					/>
 				))}
 			</div>
-		
+
```
Comment (Harshil-Jani) [inline pages/blog/index.js:59]: Please remove extra whitespaces.

File: styles/Search.module.css [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        height: 50px;
+        /* background:linear-gradient(to left, rgb(249, 116, 249), rgb(62, 151, 102)); */
+        background: purple;
+        color: white;
+        border: 0;
+        border-radius: 0 30px 30px 0;
+        font-size: 16px;
+        cursor: pointer;
+        box-shadow: 2px 2px 5px 0px grey;
+    }
```
Comment (Harshil-Jani) [inline styles/Search.module.css:34]: Add new line at EOF

File: styles/scss/blog.module.scss [hunk 8]
```diff
@@ -5,3 +5,38 @@
...
+      height: 50px;
+      /* background:linear-gradient(to left, rgb(249, 116, 249), rgb(62, 151, 102)); */
+      background: purple;
+      color: white;
+      border: 0;
+      border-radius: 0 30px 30px 0;
+      font-size: 16px;
+      cursor: pointer;
+      box-shadow: 2px 2px 5px 0px grey;
+  }
```
Comment (Harshil-Jani) [inline styles/scss/blog.module.scss:42]: New line at EOF

**Review by Harshil-Jani — CHANGES_REQUESTED** [review]

**Comment by Harshil-Jani** [comment]
@BhartiNagpure Functionality wise the search bar is not working. Can you please look into it ?

**Comment by BhartiNagpure** [comment]
> @BhartiNagpure Functionality wise the search bar is not working. Can you please look into it ?

from my side it is work also I check the deployment it work properly

**Comment by BhartiNagpure** [comment]
Screenshot 2023-07-14 135951
Screenshot 2023-07-14 140008

This is work in deployment.

**Comment by Harshil-Jani** [comment]
I see. I was trying to search with `How`
image

**Comment by Harshil-Jani** [comment]
For example there is an article titled `Learn serverless on AWS step-by-step - SQL with Aurora` and I would want to search that. If I just remember the keyword `SQL` and try to search it. I do expect in return the article to show up.
image

**Comment by BhartiNagpure** [comment]
> I see. I was trying to search with `How` image

for how it is also work
Screenshot 2023-07-14 140716

**Comment by BhartiNagpure** [comment]
@Harshil-Jani how is present in above four card and you search it not suffle but you scroll down last card is remove as it not contain how.

**Comment by BhartiNagpure** [comment]
> I see. I was trying to search with `How` image

I get it, this is because you are trying to search starting with capital letter.

**Comment by BhartiNagpure** [comment]
@Harshil-Jani you may check now it works properly for capital letter also.

File: pages/blog/index.js [hunk 9]
```diff
@@ -6,8 +6,14 @@ import styles from "@styles/scss/blog.module.scss";
...
 import WriteBlog from "./writeblog";
-
+import { useState } from "react";
 export default function Blogs() {
+	const [query, setQuery] = useState("");
+	const handleSearch=(e)=>{
+		setQuery (e.target.value)
+	}
+   const searchQueryHandler = () =>{
+   }
```
Comment (Harshil-Jani) [inline pages/blog/index.js:16]: Please remove this function `searchQueryHandler` since it is empty.

**Review by Harshil-Jani — CHANGES_REQUESTED** [review]
Great work @BhartiNagpure . Now the search works perfect. Can you please do some minor changes as commented and also please do format the code so it looks readable.

**Comment by BhartiNagpure** [comment]
> Great work @BhartiNagpure . Now the search works perfect. Can you please do some minor changes as commented and also please do format the code so it looks readable.

sure

**Comment by BhartiNagpure** [comment]
@Harshil-Jani  work done please check it.

**Comment by Harshil-Jani** [comment]
Thanks a lot @BhartiNagpure.

**Review by Harshil-Jani — APPROVED** [review]

**Comment by BhartiNagpure** [comment]
@Harshil-Jani you are not added any level , I am not getting points.

**Comment by Harshil-Jani** [comment]
Added @BhartiNagpure

**Pull request merged**